browser-extension/mv3/src/service-worker/services/messageHandler.ts

```diff
@@ -30,9 +30,14 @@ export const initMessageHandler = () => {
     /* From any case, return true when sendResponse is called asynchronously */
     switch (message.action) {
       case EXTENSION_MESSAGES.HANDSHAKE_CLIENT:
-        initClientHandler({
-          tabId: sender.tab?.id,
-          frameIds: [sender.frameId],
+        isExtensionEnabled().then((isExtensionStatusEnabled) => {
+          if (!isExtensionStatusEnabled) return;
+
+          initClientHandler({
+            tabId: sender.tab?.id,
+            frameIds: [sender.frameId],
+          });
+          applyScriptRules(sender.tab?.id, sender.frameId, sender.url);
         });
         initCustomWidgets(sender.tab?.id, sender.frameId);
         applyScriptRules(sender.tab?.id, sender.frameId, sender.url);
```

Comment:
Why are we applying ScriptRules and initializing custom Widgets in the line number 42 and 43? Shouldn't they also go inside the check whether extension is enabled or not?

Reply:
Also, we should initSessionRecording only when extension is enabled?

Reply:
The same goes for start recording explicity

Reply:
Yes. They should be. I think while resolving merge conflicts with custom widgets PR, they are duplicated outside

Reply:
Fixed it